spyderlib/spyder_restart.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Restart Spyder
+
+A helper script to allow for spyder to restart
+"""
+
+import os
+import os.path as osp
+import sys
+import platform
+import subprocess
+import time
+
+
+def main():
+    # Variables defined in spyderlib\spyder.py 'restart()' method
+    spyder_args = os.environ.pop('SPYDER_ARGS', None)
+    pid = os.environ.pop('SPYDER_PID', None)
+    is_bootstrap = os.environ.pop('SPYDER_IS_BOOTSTRAP', None)
+    spyder_folder = os.environ.pop('SPYDER_FOLDER', None)
+
+    if any([not spyder_args, not pid, not is_bootstrap, not spyder_folder]):
+        error = "This script can only be called from within a Spyder instance"
+        raise Exception(error)
+
+    # Fix variables
+    args = eval(spyder_args)
+    args = ' '.join(args)
+    is_bootstrap = eval(is_bootstrap)
+    pid = eval(pid)
+
+    # Fix args
+    # Maybe --new-instance should always be included to cope with corner cases
+    # where someone has one running version without and and another with
+    # If the user the one without (--new...) it would not start
+
+    python = sys.executable
+
+    # Build the command7024
+    if is_bootstrap:
+        spyder = osp.join(spyder_folder, 'bootstrap.py')
+    else:
+        spyderlib = osp.join(spyder_folder, 'spyderlib')
+        spyder = osp.join(spyderlib, 'start_app.py')
+
+    command = '"{0}" "{1}" {2}'.format(python, spyder, args)
+
+    # Adjust the command depending on the OS  start cmd /K " "C:\Anaconda\python.exe" "C:\Anaconda\lib\site-packages\spyderlib\spyder_restart.py" "
+    system = platform.system().lower()
+    if 'win' in system: 
+        shell = False
+        is_pid_running = _is_pid_running_on_windows
+    elif 'linux' in system:
+        command = command + ' &'
+        shell = True
+        is_pid_running = _is_pid_running_on_unix
```

Comment:
So, I can run it without the ` &` , but I need to run it with `shell=True` otherwise it does not seem to work.

I just checked and it works as expected in both installed version and bootstrap version. I will make the push tomorrow... but I still someone with a Mac... @ccordoba12 ?